lib/date_time/date_time.c

```diff
@@ -334,11 +348,6 @@ static void date_time_handler(struct k_work *work)
 {
 	if (CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS > 0) {
```

Comment:
The interval is already checked when the work is scheduled in `date_time_notify_event()`. However, the interval is not checked when the work is scheduled in `date_time_init()`, so I think if CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS is 0, the work is executed immediately. I think it would be clearer to check the interval before scheduling the work and remove the check from the work handler.

Reply:
I know but I didn't consider this such a bad thing. `date_time_init()` is not perfect at the moment because if you have default 1h interval, the first time update comes 1h after `date_time_init()` is called. Maybe this is OK and user can trigger the first date time event by itself but I think in ideal world it would come automatically, e.g., when LTE network registration has occurred. I did leave all this untouched because I wanted to put limit to what I'm doing with this PR plus I'm not sure these changes would really be needed.
I can definitely add this particular if statement to the init function if that's preferred.

Reply:
As discussed earlier today, I'll add the comparison to `date_time_init` and remove from `date_time_handler()`.